soilDB 2.9.0

brownag opened the pull request:
This will be a significant minor release of soilDB that removes several deprecated aspects of result structure, function arguments, etc. 

 - [x] Remove deprecated NASIS column aliases (https://ncss-tech.github.io/AQP/soilDB/bulletins/2025.01-1-soilDB-NASIS-column-aliases.html)
 - [x] Bump to minimum R version 4.1 #424
    - [x] Remove deprecated `stringsAsFactors` usages
 - <strike> Return original NASIS record IDs from SDM in relevant SDA queries https://github.com/ncss-tech/soilDB/issues/397</strike> Full resolution of issue 397 has been deferred pending SSURGO schema updates or new routines for amending schema
 - [x] Implement better checks of terra functionality for safe skipping of tests, examples, and checking required namespace https://github.com/ncss-tech/soilDB/issues/375
 
 
 Will close #242, #375, and #424

brownag committed:
docs: add new SDA vignette

AndrewBrown-USDA committed:
chore: upgrade minimum R to >=4.1.0

chore: bump R package version

pkgver

AndrewBrown-USDA committed:
chore(soilDB_user_dir): remove support for R <4.0

AndrewBrown-USDA committed:
feat!: remove deprecated `stringsAsFactors`

BREAKING CHANGE

wip: more stringsAsFactors fixes

fix(SSURGO_spatial_query): eliminate side effects on user options

fix: uncode

AndrewBrown-USDA committed:
fix(OSD/soilweb): donttest examples

AndrewBrown-USDA committed:
docs: update NEWS

AndrewBrown-USDA committed:
fix: do not call `library()` in tests

AndrewBrown-USDA committed:
fix(fetchSCAN): skip TZ test during site maintenance

AndrewBrown-USDA committed:
fix: remove WIP function from NAMESPACE

AndrewBrown-USDA committed:
chore: clean up stale TODO items and comments

AndrewBrown-USDA committed:
refactor(.cogmd_prep): clean up factor logic (no change in behavior)

AndrewBrown-USDA committed:
feat: add helper to check functional terra install

AndrewBrown-USDA committed:
Merge branch 'master' into rdep41

brownag committed:
refactor: datasets

brownag committed:
fix: get_mapunit_from_NASIS to use `NASISDomainsAsFactors()`

brownag committed:
fix: get_projectmapunit_from_NASIS `dsn` argument

brownag committed:
chore: bump docs for roxygen 7.3.3

brownag committed:
feat!(get_site_data_from_NASIS_db): remove aliases
 - BREAKING CHANGE

brownag committed:
feat!(fetchNASIS_components): remove aliases
 - BREAKING CHANGE

brownag committed:
feat!(get_component_data_from_NASIS_db): remove aliases
 - BREAKING CHANGE

brownag committed:
feat!(get_ecosite_history_from_NASIS_db): remove non-standard aliases, add TODO for "classifier" conflict

brownag committed:
feat!(get_hz_data_from_NASIS_db): remove aliases
 - BREAKING CHANGE

brownag committed:
feat!(get_labpedon|layer_data_from_NASIS_db): remove aliases
 - BREAKING CHANGE

brownag committed:
feat!(get_veg(plot)_data_from_NASIS_db): remove aliases and deprecate "get_veg_data_from_NASIS_db" superseded by `fetchVegdata()`

brownag committed:
doc: clean up alias references in fetchNASIS columns vignette

brownag committed:
vegplot

brownag committed:
fix: getHzErrorsNASIS use of pedon_id

brownag committed:
fix!(fetchNASIS): remove shims for `pedon_id` and `surface_fgravel`

brownag committed:
refactor: replace httr and xml2 dependencies from SDA_query with base R

AndrewBrown-USDA committed:
fix: add numeric equivalents for unescaping XML

AndrewBrown-USDA committed:
feat: add custom error for maintenance exception
 - the "daily maintenance" response is not a standard JSON/XML service exception, just an HTML page

AndrewBrown-USDA committed:
build(deps): move jsonlite to Imports

AndrewBrown-USDA committed:
Merge branch 'master' into rdep41

brownag committed:
Merge remote-tracking branch 'origin/rdep41' into rdep41

brownag committed:
fix(tests): fetchSCAN allow for setups that use UTC timezone always

brownag committed:
fix: cleanup R objects

brownag committed:
Merge remote-tracking branch 'origin/rdep41' into rdep41

AndrewBrown-USDA committed:
doc: update NEWS

AndrewBrown-USDA committed:
Revert "feat: add helper to check functional terra install"

This reverts commit 4e2bb50b2835bcfc450389911996279f6622f7d5.

AndrewBrown-USDA committed:
fix: check for functional terra proj for #375

AndrewBrown-USDA committed:
Merge branch 'master' into rdep41

dylanbeaudette commented on vignettes/sda.Rmd:
````diff
@@ -0,0 +1,2100 @@
+
+```sql
+/* Exclude "Duff" / Dry Organics */
+AND ISNULL(horztype, '') <> 'Dry Organic'
````
Consider using all `!=` or `<>`, I see both used in this document. Even though TSQL suggests `<>`, `!=` works fine and is closer to what we use in R.

brownag committed:
doc: replace some SQL `<>` with `!=`

brownag committed:
doc: cleanup sda vignette

AndrewBrown-USDA committed:
docs: fix broken URLs for CRAN

AndrewBrown-USDA committed:
chore: remove deprecated internal functions

AndrewBrown-USDA committed:
doc: use functional iterators and give an SSA chunk example

AndrewBrown-USDA committed:
doc: add an SDA error handling section

AndrewBrown-USDA committed:
docs: more URL fixes

AndrewBrown-USDA committed:
doc: raca URL

AndrewBrown-USDA committed:
docs: a couple more URL issues form `tools::check_package_urls()`

AndrewBrown-USDA committed:
fix(get_SDA_property): add match.arg support for shortcuts for method="MIN" or "MAX"
 - this has always worked historically, but  the recent attempt to provide more informative error message omitted these

AndrewBrown-USDA committed:
docs: sda vignette adjustments

AndrewBrown-USDA committed:
doc: update NEWS
